ethcore/src/state/mod.rs

```diff
@@ -43,41 +44,86 @@ pub struct ApplyOutcome {
 pub type ApplyResult = Result<ApplyOutcome, Error>;
 
 #[derive(Debug)]
-enum AccountEntry {
-	/// Contains account data.
-	Cached(Account),
-	/// Account has been deleted.
-	Killed,
-	/// Account does not exist.
-	Missing,
+/// Account data. Can hold existing account or record that it does not exist.
+enum MaybeAccount {
+	/// Existing account.
+	Some(Account),
+	/// Account does not exist (missing or deleted).
+	None,
 }
 
-impl AccountEntry {
-	fn is_dirty(&self) -> bool {
+impl MaybeAccount {
```

Comment:
or even just swap `.clone_dirty()` for `.as_ref().map(Account::clone_dirty)` even?